modules/axhal/src/trap.rs

```diff
@@ -1,23 +1,31 @@
 //! Trap handling.
 
-use crate_interface::{call_interface, def_interface};
+use linkme::distributed_slice as def_trap_handler;
+use memory_addr::VirtAddr;
+use page_table_entry::MappingFlags;
 
-/// Trap handler interface.
-///
-/// This trait is defined with the [`#[def_interface]`][1] attribute. Users
-/// should implement it with [`#[impl_interface]`][2] in any other crate.
-///
-/// [1]: crate_interface::def_interface
-/// [2]: crate_interface::impl_interface
-#[def_interface]
-pub trait TrapHandler {
-    /// Handles interrupt requests for the given IRQ number.
-    fn handle_irq(irq_num: usize);
-    // more e.g.: handle_page_fault();
-}
+pub use linkme::distributed_slice as reg_trap_handler;
```

Comment:
what does 'reg' in `reg_trap_handler` mean?

Reply:
register